CMakeLists.txt

```diff
@@ -300,6 +300,7 @@ if(ENABLE_ASM)
 		elseif(CMAKE_SYSTEM_NAME STREQUAL "SunOS" AND "${CMAKE_SYSTEM_PROCESSOR}" STREQUAL "i386")
 			set(HOST_ASM_ELF_X86_64 true)
 		endif()
+		set(CMAKE_ASM_FLAGS "${ASM_FLAGS} -Wa,--noexecstack")
```

Comment:
Defining HAVE_GNU_STACK would also work here. Should we go with belt-and-suspenders, or even modify autoconf to prefer -Wa,--noexecstack?

Reply:
Oh I totally missed the autoconf check. We can just reuse HAVE_GNU_STACK in cmake to reduce the diff.

Reply:
Thanks @tobhe!